ocrd_tesserocr/binarize.py

```diff
@@ -6,35 +6,26 @@
     PSM, RIL
 )
 
-from ocrd_utils import (
-    getLogger,
-    assert_file_grp_cardinality,
-    make_file_id,
-    MIMETYPE_PAGE
-)
-from ocrd_modelfactory import page_from_file
 from ocrd_models.ocrd_page import (
     AlternativeImageType,
     TextRegionType,
-    to_xml
 )
 
-from .config import OCRD_TOOL
 from .recognize import TesserocrRecognize
 
-TOOL = 'ocrd-tesserocr-binarize'
-
 class TesserocrBinarize(TesserocrRecognize):
-    def __init__(self, *args, **kwargs):
-        kwargs.setdefault('ocrd_tool', OCRD_TOOL['tools'][TOOL])
-        super().__init__(*args, **kwargs)
-        if hasattr(self, 'parameter'):
-            self.logger = getLogger('processor.TesserocrBinarize')
+    @property
+    def executable(self):
+        return 'ocrd-tesserocr-binarize'
+
+    def _init(self):
+        # use default model (eng) with vanilla tesserocr API
+        self.tessapi = PyTessBaseAPI()
 
-    def process(self):
+    def process_page_pcgts(self, pcgts, output_file_id=None, page_id=None):
```

Comment:
Besides missing the typing, this has a different signature, pcgts is not variadic here. Python allows this and it is convenient but I am wondering whether it would be better to be consistent with the typing?

```suggestion
    def process_page_pcgts(self, *input_pcgts, output_file_id=None, page_id=None):
        pcgts = input_pcgts[0]
```

Reply:
Typing – sure, I was just to lazy again.

Variadic – I thought it would be clearer like that. So we would not have to do any arity checking in the function itself – a type checker could simply detect invalid use cases which do pass multiple pages at once. But perhaps I am wrong. (Also, we already have the arity assertion in `setup`.)